src/main/java/com/google/devtools/build/lib/remote/options/CommonRemoteOptions.java

```diff
@@ -33,4 +38,39 @@ public class CommonRemoteOptions extends OptionsBase {
               + " the client to request certain artifacts that might be needed locally (e.g. IDE"
               + " support)")
   public List<String> remoteDownloadRegex;
+
+  @Option(
+      name = "experimental_remote_cache_age",
+      defaultValue = "2h",
+      documentationCategory = OptionDocumentationCategory.REMOTE,
+      effectTags = {OptionEffectTag.EXECUTION},
+      converter = RemoteDurationConverter.class,
+      help = "The guaranteed minimal age of blobs in the remote cache after their digests are "
+          + "recently referenced e.g. by an ActionResult. Bazel does several optimizations based on "
+          + "the blobs' age e.g. doesn't repeatedly call GetActionResult in an incremental build. "
+          + "The value should be set slightly less than the real age since there is a gap between "
+          + "when the server returns the digests and when Bazel receives them."
```

Comment:
I think this should say what setting this to `0` does. Also, I think the default should be `0`.

Reply:
I will add more details and what the behaviour of setting this to `0` after I nail down all the implementation details.

I agree `2h` is probably not a good default but `0` isn't neither. Probably we should default to `null` which means Bazel assume the blobs will never expire (just like what it does today).

Reply:
Setting to `0` will have extremely bad incremental build performance: Bazel have to invalidate all in-memory cache and call `GetActionResult` on every action.

Reply:
Once per build, right?

If it can't be 0, then a low non-zero value is better than null. In that case I'm supportive of a couple hours. Maybe have it default to whatever the value of https://bazel.build/reference/command-line-reference#flag--max_idle_secs is?

Reply:
> Once per build, right?

yes, once per action per build.



> If it can't be 0, then a low non-zero value is better than null. In that case I'm supportive of a couple hours. Maybe have it default to whatever the value of https://bazel.build/reference/command-line-reference#flag--max_idle_secs is?

Not sure. I will find a good default value once this PR is finalized.